File: backend/src/main/java/com/realestate/agent/service/UserService.java
package com.realestate.agent.service;

import com.realestate.agent.dto.UserRequest;
import com.realestate.agent.dto.UserResponse;

import java.util.List;

public interface UserService {

    List<UserResponse> getAllUsers();

    UserResponse getUserById(Long userId);

    UserResponse createUser(UserRequest request);

    UserResponse updateUser(Long userId, UserRequest request);

    UserResponse toggleUserStatus(Long userId);

    void deleteUser(Long userId);
}

File: frontend/real-estate-frontend/src/services/userService.js
import apiClient from "./apiClient";

// Get all registered users from PostgreSQL
export const getAllUsers = async () => {
  return apiClient.get("/api/admin/users");
};

// Get user by ID
export const getUserById = async (userId) => {
  return apiClient.get(`/api/admin/users/${userId}`);
};

// Create a new user
export const createUser = async (userData) => {
  return apiClient.post("/api/admin/users", userData);
};

// Update existing user
export const updateUser = async (userId, userData) => {
  return apiClient.put(`/api/admin/users/${userId}`, userData);
};

// Toggle user active/inactive status
export const toggleUserStatus = async (userId) => {
  return apiClient.patch(`/api/admin/users/${userId}/status`);
};

// Delete user by ID
export const deleteUser = async (userId) => {
  return apiClient.delete(`/api/admin/users/${userId}`);
};
